libfreerdp/core/settings.c

```diff
@@ -1324,34 +1324,39 @@ static BOOL freerdp_settings_int_buffer_copy(rdpSettings* _settings, const rdpSe
 	if (!_settings || !settings)
 		return FALSE;
 
+	typedef struct
 	{
-		const void* data = freerdp_settings_get_pointer(settings, FreeRDP_LoadBalanceInfo);
-		const UINT32 len = freerdp_settings_get_uint32(settings, FreeRDP_LoadBalanceInfoLength);
-		if (!freerdp_settings_set_pointer_len(_settings, FreeRDP_LoadBalanceInfo, data, len))
-			return FALSE;
-	}
-	{
-		const void* data = freerdp_settings_get_pointer(settings, FreeRDP_ServerRandom);
-		const UINT32 len = freerdp_settings_get_uint32(settings, FreeRDP_ServerRandomLength);
-		if (!freerdp_settings_set_pointer_len(_settings, FreeRDP_ServerRandom, data, len))
-			return FALSE;
-	}
+		FreeRDP_Settings_Keys_UInt32 lenKey;
+		FreeRDP_Settings_Keys_Pointer pointerKey;
+	} PointerDefs;
+
+	PointerDefs pointerDefs[] = {
+		/* */
+		{ FreeRDP_LoadBalanceInfoLength, FreeRDP_LoadBalanceInfo },
+		{ FreeRDP_ServerRandomLength, FreeRDP_ServerRandom },
+		{ FreeRDP_ClientRandomLength, FreeRDP_ClientRandom },
+		{ FreeRDP_ServerCertificateLength, FreeRDP_ServerCertificate },
+		{ FreeRDP_ChannelDefArraySize, FreeRDP_ChannelDefArray },
+		{ FreeRDP_NumMonitorIds, FreeRDP_MonitorIds },
+		{ FreeRDP_BitmapCacheV2NumCells, FreeRDP_BitmapCacheV2CellInfo },
+		{ FreeRDP_RedirectionPasswordLength, FreeRDP_RedirectionPassword },
+		{ FreeRDP_RedirectionTsvUrlLength, FreeRDP_RedirectionTsvUrl },
+		{ FreeRDP_RedirectionGuidLength, FreeRDP_RedirectionGuid }
+	};
+
+	for (int i = 0; i < ARRAYSIZE(pointerDefs); i++)
 	{
-		const void* data = freerdp_settings_get_pointer(settings, FreeRDP_ClientRandom);
-		const UINT32 len = freerdp_settings_get_uint32(settings, FreeRDP_ClientRandomLength);
-		if (!freerdp_settings_set_pointer_len(_settings, FreeRDP_ClientRandom, data, len))
-			return FALSE;
+		UINT32 n = freerdp_settings_get_uint32(settings, pointerDefs[i].lenKey);
+
+		const void* ptr = freerdp_settings_get_pointer(settings, pointerDefs[i].pointerKey);
```

Comment:
maybe extract `pointerDefs[i]` to a loop variable so it is easier inspected in debuggers?